src/pybind/mgr/dashboard/frontend/src/app/shared/directives/auth-storage.directive.ts

```diff
@@ -0,0 +1,63 @@
+import { Directive, Input, TemplateRef, ViewContainerRef } from '@angular/core';
+
+import _ from 'lodash';
+
+import { Permissions } from '~/app/shared/models/permissions';
+import { AuthStorageService } from '~/app/shared/services/auth-storage.service';
+
+@Directive({
+  selector: '[cdScope]'
+})
+export class AuthStorageDirective {
+  permissions: Permissions;
+
+  constructor(
+    private templateRef: TemplateRef<any>,
+    private viewContainer: ViewContainerRef,
+    private authStorageService: AuthStorageService
+  ) {}
+
+  @Input('cdScope') set cdScope(condition: string | string[] | object) {
+    this.permissions = this.authStorageService.getPermissions();
+    if (this.isAuthorized(condition)) {
+      this.viewContainer.createEmbeddedView(this.templateRef);
+    } else {
+      this.viewContainer.clear();
+    }
+  }
+
+  @Input() cdScopeMatchAll = false;
+
+  private isAuthorized(condition: any): boolean {
```

Comment:
```suggestion
  private isAuthorized(condition: string | string[] | object): boolean {
```

Reply:
Another iteration on @alfonsomthd comment (a tricky one):
```suggestion
  private isAuthorized(condition: string | string[] | Partial<{[Property in keyof Permissions]: keyof Permission}> ): boolean {
```

Reply:
@epuertat please explain more. This really looks tricky

Reply:
In this case the type `[Property in keyof Permissions]: Array<keyof Permissions>]` is syntactic sugar for a mapping type like `[key: 'pool' | 'osd' | 'osd' ...]: Array<'create' | 'update' ...>`. And `Partial<>` is needed because the condition `cdScope="{pool: ['read']}"` is valid and doesn't have all the scopes defined: `pool, rgw, etc`.